MM-30494. Added configuration with openLDAP.

rachelbritton opened the pull request:
Added section on configuring LDAP with openLDAP.

rachelbritton committed:
MM-30494. Added configuration with openLDAP.

Added section on configuring LDAP with openLDAP.

ddespic commented on guides/implementation/ldap_user_auth.rst:
```diff
@@ -118,6 +118,65 @@ Example configuration for connecting to an AD LDAP service:
+    "method": "authenticate",
+    "server": {
+        "uri": "ldap://ldap.example.com:636",
+        "reader_dn": "cn=ldap_users,dc=dev,dc=lab",
```
reader_dn is a path in LDAP to the admin of the LDAP (if LDAP is not open to everyone, the admin of the LDAP has to authenticate) so here cn is the username of an LDAP admin and one below is his password
You can write here cn=admin insted of cn=ldap_users

rachelbritton replied:
@ddespic The example for AD config uses "dc=corp, dc=example, dc=com". Should we use that as well here instead of dc=dev, dc=lab?

ddespic replied:
sure, it is good to be consistent, this dc=dev, dc=lab path was from my setup.

ddespic commented on guides/implementation/ldap_user_auth.rst:
```diff
@@ -118,6 +118,65 @@ Example configuration for connecting to an AD LDAP service:
+    }
+
+.. note::
+  ``cn`` ("common name") is the equivalent of ``userPrincipalName`` in Active Directory (AD). Therefore, when configuring with openLDAP, ``group_search_config.name_attribute`` is the user's name, whereas it calls for the group name when configuring with AD.
```
User's name in LDAP is written in cn attribute field, also group name can be configured that way. The group name in our case is written in cn attribute field. Micetro to know where to check the group name. So in ldapconfig.json file "name_attribute" we need to write not the name of a group but the name of the attribute itself, in this case "cn"

rachelbritton committed:
MM-30494. Corrections to LDAP config file.

Made corrections to LDAP config example and following note as per feedback.

ddespic commented on guides/implementation/ldap_user_auth.rst:
```diff
@@ -118,6 +118,65 @@ Example configuration for connecting to an AD LDAP service:
+    }
+
+.. note::
+  In the ``name_attribute`` field, ``cn`` ("common name") refers to the user's name or group name in LDAP. In the LDAP configuration JSON file, enter ``cn`` in the ``name_attribute`` field, not the user or group name itself.
```
In the ``name_attribute`` field, ``cn`` ("common name") refers to the group name in LDAP. In the LDAP configuration JSON file, enter ``cn`` in the ``name_attribute`` field if that attribute is used for group name in LDAP, not the group name itself.

rachelbritton committed:
MM-30494. Changed config example path.

Changed the path in the example to match our AD example and revised the note text.

ddespic commented on guides/implementation/ldap_user_auth.rst:
```diff
@@ -118,6 +118,65 @@ Example configuration for connecting to an AD LDAP service:
+.. code-block:: bash
+
+  sudo cp ldapconf.json /var/mmsuite/mmcentral
+  sudo sed -i "s/admin_password/$PSSWD/g" /var/mmsuite/mmcentral/ldapconf.json
```
@rachelbritton maybe we can skip this "sudo sed -i ..." line. This line replaces admin_password string (like a placeholder) with real admin password stored in $PSSWD variable. Maybe we can just write that they need to replace reader_dn and reader_password values with admin credentials from LDAP.

rachelbritton committed:
MM-30494. Adjusted ldap config json.

Adjusted ldap json config file and added note about reader_dn and reader_password values, as per feedback.
